### PR TITLE
[PyTorch] Fix ONNX/dynamo error

### PR DESCRIPTION
**Pull request by cyanguwa**
This PR fixes the dynamo error when running ONNX tests: `RuntimeError: Detected that you are using FX to torch.jit.trace a dynamo-optimized function.`

### CODE REVIEW

**Comment by ptrendx** [comment]
I would actually prefer this to go in `transformer_engine/pytorch/__init__.py` (with some `try/except` block to make sure that it will not fail in an older pytorch version).

**Comment by cyanguwa** [comment]
@ptrendx let me know if [4479d5f](https://github.com/NVIDIA/TransformerEngine/pull/562/commits/4479d5f4a578794199524d1af09a36c499088737) looks ok to you. Thanks!

File: transformer_engine/pytorch/__init__.py [hunk 1]
```diff
@@ -29,3 +29,9 @@
     onnx_rmsnorm_fwd,
     onnx_rmsnorm_fwd_fp8
 )
+try:
+    import torch
+    from .tests.pytorch import seed_default_rng, set_max_seq_len, test_export_gpt_generation
+    test_export_gpt_generation(seed_default_rng, set_max_seq_len, False, precision1, True)
+except:
+    torch._dynamo.config.error_on_nested_jit_trace = False
```
Comment (ptrendx) [inline transformer_engine/pytorch/__init__.py:37]: Uhhh, actually what I had in mind is this:

```suggestion
try:
    import torch
    torch._dynamo.config.error_on_nested_jit_trace = False
except:
    pass
```

Comment (cyanguwa) [reply transformer_engine/pytorch/__init__.py]: Sounds good

**Pull request merged**